plugins/async-node/core/src/index.test.ts

```diff
@@ -340,3 +342,64 @@ test('replaces async nodes with chained multiNodes singular', async () => {
   expect(view?.actions[1].asset.type).toBe('text');
   expect(view?.actions[2].asset.type).toBe('text');
 });
+
+test('should call onAsyncNode hook when async node is encountered', async () => {
+  const plugin = new AsyncNodePlugin();
+
+  let deferredResolve: ((value: any) => void) | undefined;
+
+  // Call the tap method of the onAsyncNode hook
+ plugin.hooks.onAsyncNode.tap('test', async (node: Node.Async) => {
+    return new Promise((resolve) => {
+      deferredResolve = resolve;
+    });
+  });
+
+  let updateNumber = 0;
+
+  const player = new Player({ plugins: [plugin] });
+
+  player.hooks.viewController.tap('async-node-test', (vc) => {
+    vc.hooks.view.tap('async-node-test', (view) => {
+      view.hooks.onUpdate.tap('async-node-test', (update) => {
+        updateNumber++;
+      });
+    });
+  });
+
+  player.start(basicFRFWithActions as any);
+
+  let view = (player.getState() as InProgressState).controllers.view.currentView
+    ?.lastUpdate;
+
+  expect(view).toBeDefined();
+  expect(view?.actions[1]).toBeUndefined();
+
+  await waitFor(() => {
+    expect(updateNumber).toBe(1);
+    expect(deferredResolve).toBeDefined();
+  });
+
+  // Call the deferredResolve with type 'Async'
+  if (deferredResolve) {
```

Comment:
this is likely not necessary for the reason stated above, we don't need to resolve the async node with anything for this test, we just need to check the tap matches what's [here](https://github.com/player-ui/player/blob/main/plugins/async-node/core/src/index.ts#L56-L63) along with the properties

Reply:
redone the entire test case